src/main/java/uk/ac/sanger/sccp/stan/service/label/LabwareLabelDataService.java

```diff
@@ -56,6 +61,56 @@ public LabwareLabelData toLabelData(Labware labware, List<LabelContent> content,
         return new LabwareLabelData(labware.getBarcode(), medium, dateString, content);
     }
 
+    /**
+     * Label data where we list the tissue for each row.
+     * @param labware the labware the label describes
+     * @return label data describing the labware
+     */
+    public LabwareLabelData getRowBasedLabelData(Labware labware) {
+        // 1. Load simple contents into a map.
+        // 2. Check contents are suitable.
+        // 3. Convert to correct number of label contents.
+        Map<Address, List<SimpleContent>> map = addressToSimpleContent(labware);
+        if (map.isEmpty()) {
+            return toLabelData(labware, List.of(), Set.of());
+        }
+        Tissue[] tissues = checkRowBasedLayout(labware, map);
+
+        final int numTissues = tissues.length;
+        String[] donorNames = new String[numTissues];
+        String[] tissueDescs = new String[numTissues];
+        String[] reps = new String[numTissues];
+        Set<String> mediums = new HashSet<>(numTissues);
+
+        for (int i = 0; i < numTissues; i++) {
+            Tissue tissue = tissues[i];
+            if (tissue != null) {
+                donorNames[i] = tissue.getDonor().getDonorName();
+                tissueDescs[i] = getTissueDesc(tissue);
+                reps[i] = tissue.getReplicate();
+                mediums.add(tissue.getMedium().getName());
+            }
+        }
+
+        List<LabelContent> content = new ArrayList<>(numTissues);
+        final int numCols = labware.getLabwareType().getNumColumns();
+        for (int i = 0; i < numTissues; ++i) {
+            Tissue tissue = tissues[i];
+            if (tissue==null) {
+                content.add(new LabelContent(null, null, null));
+            } else {
+                final int row = i+1;
+                Stream<SimpleContent> scStream =  IntStream.range(1, numCols+1)
+                        .mapToObj(col -> map.get(new Address(row, col)))
+                        .filter(Objects::nonNull)
+                        .flatMap(Collection::stream);
+                Integer[] sectionRange = sectionRange(scStream.iterator());
+                content.add(new LabelContent(donorNames[i], tissueDescs[i], reps[i], sectionRange[0], sectionRange[1]));
+            }
```

Comment:
Just a doubt, in story , it says 1 donor per row, no section number, just donor and tissue information on the label. 
Is section range also required?

Reply:
The code for extracting label data is decoupled from the label template itself. It loads all the fields it can, even if the template does not use all of them.

Reply:
Ok. Thank you.